packages/react-accordion/src/components/Accordion/useAccordionContext.ts

```diff
@@ -0,0 +1,106 @@
+import * as React from 'react';
+import { createDescendantContext, useDescendant, useDescendantsInit } from '../../utils/descendants';
+import { AccordionContext, AccordionDescendant, AccordionIndex, AccordionState } from './Accordion.types';
+import { useConst, useControllableValue, useEventCallback } from '@fluentui/react-utilities';
+import { createContext } from '@fluentui/react-context-selector';
+
+export const accordionDescendantContext = createDescendantContext<AccordionDescendant>('AccordionDescendantContext');
+
+export const accordionContext = createContext<AccordionContext>({
+  openItems: [],
+  requestToggle() {
+    /* noop */
+  },
+});
+
+/**
+ * Creates the context to be provided for AccordionItem components
+ */
+export function useCreateAccordionContext(state: AccordionState) {
+  const { index, multiple, collapsible, onToggle, size, expandIcon, expandIconPosition, button } = state;
+  const isControlled = useConst(typeof index !== 'undefined');
+  const [descendants, setDescendants] = useDescendantsInit<AccordionDescendant>();
+  // eslint-disable-next-line react-hooks/exhaustive-deps
+  const initialDefaultIndex = React.useMemo(() => initializeUncontrolledOpenItems(state), []);
+  const normalizedIndex = React.useMemo(() => (index !== undefined ? normalizeIndex(index) : undefined), [index]);
+  const [openItems, setOpenItems] = useControllableValue(normalizedIndex, initialDefaultIndex);
```

Comment:
As #17293 have been merged, can we use a function there?

```suggestion
  const [openItems, setOpenItems] = useControllableValue(normalizedIndex, () => initializeUncontrolledOpenItems(state));
```

Reply:
Oh, that's actually the reason I asked @ling1726  for these initializer feature! I'll actually make some changes than, there's more than just the initializer!